src/server/img_storage.rs

```diff
@@ -214,3 +418,133 @@ impl StoreImage {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::settings::test_settings;
+    use actix_http::http::Uri;
+    use rand::Rng;
+
+    fn set_env() {
+        // sometimes the IDE doesn't include the host env vars.
+        // this lets you set them proactively.
+
+        // std::env::set_var("GOOGLE_APPLICATION_CREDENTIALS","/home/jrconlin/.ssh/keys/sync-spanner-dev.json");
+    }
+
+    fn test_storage_settings() -> StorageSettings {
+        let project =
+            std::env::var("CONTILE_TEST_PROJECT").unwrap_or_else(|_| "topsites_nonprod".to_owned());
+        let bucket =
+            std::env::var("CONTILE_TEST_BUCKET").unwrap_or_else(|_| "moz_test_bucket".to_owned());
+        let cdn = std::env::var("CONTILE_TEST_CDN_HOST")
+            .unwrap_or_else(|_| "https://example.com".to_owned());
+        StorageSettings {
+            project_name: project,
+            bucket_name: bucket,
+            bucket_ttl: 86400 * 15,
+            cache_ttl: 86400 * (15 + 1),
+            cdn_host: cdn,
+            ..Default::default()
+        }
+    }
+
+    fn test_image_buffer(height: u32, width: u32) -> Bytes {
+        let mut rng = rand::thread_rng();
+        // generate a garbage image.
+        let img = image::ImageBuffer::from_fn(width, height, |_x, _y| {
+            image::Rgb([
+                rng.gen_range(0..255),
+                rng.gen_range(0..255),
+                rng.gen_range(0..255),
+            ])
+        });
+        let buf: Vec<u8> = Vec::new();
+        let mut out = std::io::Cursor::new(buf);
+        let mut encoder = image::codecs::jpeg::JpegEncoder::new(&mut out);
+        encoder
+            .encode(&img.into_raw(), width, height, image::ColorType::Rgb8)
+            .unwrap();
+        Bytes::from(out.into_inner())
+    }
+
+    #[test]
+    fn test_config() {
+        let test_val = r#"{"project_name": "project", "bucket_name": "bucket"}"#;
+
+        let mut setting = test_settings();
+        setting.storage = test_val.to_owned();
+        let store_set: StorageSettings = (&setting).into();
+
+        assert!(store_set.project_name == *"project");
+        assert!(store_set.bucket_name == *"bucket");
+    }
+
+    #[tokio::test]
+    async fn test_image_proc() -> Result<(), ()> {
+        if std::env::var("GOOGLE_APPLICATION_CREDENTIALS").is_err() {
+            print!("Skipping test: No credentials found.");
+            return Ok(());
+        }
+        let src_img = "https://evilonastick.com/test/128px.jpg";
+
+        let test_settings = test_storage_settings();
+        let bucket = StoreImage::build_bucket(&test_settings)
+            .await
+            .unwrap()
+            .unwrap();
+        let target = src_img.parse::<Uri>().unwrap();
+        let result = bucket.store(&target).await;
+        assert!(result.is_ok());
```

Comment:
debugging nit:

This assert ends up hiding the error, this'll show it:

```suggestion
        bucket.store(&target).await.expect("Store failed");
```